ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/deduplication/CommandDeduplicationBase.scala

```diff
@@ -469,7 +470,52 @@ private[testtool] abstract class CommandDeduplicationBase(
 
 object CommandDeduplicationBase {
   trait DelayMechanism {
-    def apply(): Future[Unit]
+    val deduplicationDuration: Duration
+    val extraWait: Duration
+
+    /** Delay by the guaranteed full deduplication period. After calling this method any duplicate calls should succeed
+      */
+    def delayForEntireDeduplicationPeriod(): Future[Unit] = delayBy(
+      deduplicationDuration + extraWait
+    )
```

Comment:
Tiny nitpick: I find one-liners with long declarations easier to read if you put the line break after the `=`.

This is unfortunately something scalafmt can't do, so I am not so fussed that I would block anything; just bringing it up.

```suggestion
    def delayForEntireDeduplicationPeriod(): Future[Unit] =
      delayBy(deduplicationDuration + extraWait)
```

Reply:
Fully agree with you, reorganized the methods.